ui/controls.js

```diff
@@ -587,21 +587,51 @@ shaka.ui.Controls = class extends shaka.util.FakeEventTarget {
     return false;
   }
 
+  /** @private */
+  async enterFullScreen_() {
+    try {
+      if (document.fullscreenEnabled) {
+        await this.videoContainer_.requestFullscreen({navigationUI: 'hide'});
+      } else {
+        const video = /** @type {HTMLVideoElement} */(this.localVideo_);
+        if (video.webkitSupportsFullscreen) {
+          video.webkitEnterFullscreen();
+        }
+      }
+    } catch (error) {
+      // Entering fullscreen can fail
+      // if the user didn't interacting with the video.
+      // be rejected with an error. Suppress that error.
+    }
+  }
+
+  /** @private */
+  async exitFullScreen_() {
+    if (document.fullscreenEnabled) {
+      await document.exitFullscreen();
+    } else {
+      const video = /** @type {HTMLVideoElement} */(this.localVideo_);
+      if (video.webkitSupportsFullscreen) {
+        video.webkitExitFullscreen();
+      }
+    }
+  }
+
   /** @export */
   async toggleFullScreen() {
     if (document.fullscreenEnabled) {
```

Comment:
A lot of the logic in toggleFullScreen is now redundant, and could be rewritten in terms of enterFullScreen_(), exitFullScreen_(), and isFullScreenEnabled().